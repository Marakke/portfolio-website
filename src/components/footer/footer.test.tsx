import React from "react";
import { render } from "@testing-library/react";
import * as Gatsby from "gatsby";
import Footer from "./index";

const useStaticQuery = jest.spyOn(Gatsby, `useStaticQuery`);

const mockUseStaticQuery = {
  footer: {
    accessibilityReportTitle: "Accessibility Report Title",
    accessibilityReport: {
      url: "accessibility-report-url",
    },
    sourceCodeTitle: "Source Code Title",
    sourceCodeLink: "source-code-link",
  }
};

describe("Footer", () => {
  beforeEach(() => {
    useStaticQuery.mockImplementation(() => mockUseStaticQuery);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("renders the Footer component correctly", () => {
    const { container, getByTestId } = render(<Footer />);
    expect(container).toMatchSnapshot();

    const footerContainer = getByTestId("footer-container");
    expect(footerContainer).toBeInTheDocument();

    const anchorElements = container.querySelectorAll("a");
    expect(anchorElements.length).toEqual(2);

    expect(anchorElements[0]).toHaveTextContent("Accessibility Report Title");
    expect(anchorElements[1]).toHaveTextContent("Source Code Title");

    expect(anchorElements[0].href).toBe("http://localhost/accessibility-report-url");
    expect(anchorElements[1].href).toBe("http://localhost/source-code-link");
  });
});
