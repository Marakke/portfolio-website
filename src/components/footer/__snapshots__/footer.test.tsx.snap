// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the Footer component correctly 1`] = `
<div>
  <div
    class="footer-container"
    data-testid="footer-container"
  >
    <a
      class="accessibility-report"
      href="accessibility-report-url"
      target="_blank"
    >
      <img
        alt="Download Icon"
        src="test-file-stub"
      />
      Accessibility Report Title
    </a>
    <a
      class="source-code"
      href="source-code-link"
      target="_blank"
    >
      Source Code Title
    </a>
  </div>
</div>
`;
